### ISSUE TITLE
Change getreward(env) -> getreward(s, a, o, env)

### ISSUE THREAD

**Issue by colinxs** [Lyceum/LyceumBase.jl#2]
Kendall brought up a good point: don't want to have to compute or cache s/a/o. Keep env for convenience and parameters of env/reward function

**Comment by colinxs** [comment]
So I partially implemented it and it's kind of...messy? The nice thing about `getreward(env)` is that it's very clear what the reward is a function of. Doing `getreward(s, a, o, env)` makes it possible to ignore s/a/o and use env, leading to possibly hard to track bugs.

If the main motivation here is to avoid expensive recomputation, then perhaps the environment should just cache that result like:
```julia
function getreward(myenv)
  if env_has_changed_since_last_call()
    myenv.buf = expensive_call()
  end
  _getreward(myenv.buf)
end
```
But having to do this for every environment is also not ideal. I'm thinking we can tie this in with #3 somehow and have the best of both worlds? `BufferedEnv` or something similar could handle the stale caching logic and buffer preallocation.

**Comment by colinxs** [comment]
fixed in #7
